src/components/input-file/input-file.js

```diff
@@ -0,0 +1,136 @@
+import Component from '../../app/js/base/Component';
+
+class InputFile extends Component {
+    constructor(element) {
+        super(element);
+        this.eventName = this.root.dataset.event;
+        this.inputFile = this.getElement('input');
+        this.fileError = this.getElement('error');
+        this.inputFile.addEventListener('input',this.handleInputFile);
+    }
+
+    handleInputFile = (event) => {
+        this.hideInputFilesError();
+        this.removeFileInfoNodes();
+        const files = event.target.files;
+        let correctSize = true;
+        let correctNumber = false;
+        if (files.length>2) {
+            this.showInputFilesError('Максимум 2 файла');
+        } else correctNumber = true;
+
+        for (const file of files){
+            if (file.size>5000000) {
+                this.showInputFilesError('Максимальный размер файла не более 5мб');
+                correctSize = false;
+                break;
+            }
+        }
+        
+        const fileInputIsCorrect = correctSize && correctNumber;
+
+        if (fileInputIsCorrect) {
+            this.inputFile.dispatchEvent(new CustomEvent(this.eventName,{
+                detail: {
+                    file:true
+                }
+            }));
+            for (const file of files) {
+                const fileSize = file['size'];
+                const fileExtension = file['name'].split('.')[1];
+                this.appendFileInfo(fileExtension,fileSize);
+            }
```

Comment:
Можно немного уменьшить количество кода, применив [деструктуризацию](https://learn.javascript.ru/destructuring-assignment#destrukturizatsiya-obekta):
```suggestion
            for (const { size, name } of files) {
                const extension = name.substring(name.lastIndexOf('.') + 1);
                this.appendFileInfo(extension, size);
            }
```